xlgui/widgets/playlist.py

```diff
@@ -1295,17 +1299,38 @@ def on_drag_data_received(self, widget, context, x, y, selection, info, etime):
                         self.playlist[insert_position:insert_position] = tracks
                     else:
                         self.playlist.extend(tracks)
-
-                    return
+                        insert_position = len(self.playlist) - len(tracks)
                 except Exception:
                     pass  # Any exception on hack is ignored
+                else:
+                    # Select inserted items
+                    if tracks:
+                        self.get_selection().select_range(
```

Comment:
Seems like you should unselect prior selected tracks?

Reply:
You are probably right, but it seems to be working (maybe it unselects or moves cursor when model changes?).

Reply:
No, the old tracks are selected along with the newly dragged tracks, which isn't very intuitive.

Reply:
I think if you unselect tracks before you select new tracks, this PR is ready to go.

Reply:
> No, the old tracks are selected along with the newly dragged tracks, which isn't very intuitive.

Are you saying that you actually have this behavior?